sway-lib-std/src/tx.sw

```diff
@@ -302,7 +302,11 @@ pub fn tx_witness_data_length(index: u64) -> u64 {
 /// }
 /// ```
 pub fn tx_witness_data<T>(index: u64) -> T {
-    __gtf::<raw_ptr>(index, GTF_WITNESS_DATA).read::<T>()
+    if __size_of::<T>() == 1 {
+        __gtf::<raw_ptr>(index, GTF_WITNESS_DATA).add::<u8>(7).read::<T>()
```

Comment:
Thanks for extending the tests @xunilrj! A concern that goes through my head right now, and it's essentially the same one that you posted in the memory layout issue - the programmers who use, in this case, `__gtf`, need to be aware of the memory layout of the data they are accessing.

`__gtf` might be used in application code and there are quite some fields that return `raw_ptr`. I don't have a right feeling if a similar special case handling also has to be done in other cases, and if that might cause issues and strange bugs in applications that use `__gtf`. It has nothing to do with this PR, of course, just a general concern.

Reply:
That's one more issue to record for when we'll seek to make contracts more upgradable, but yeah I don't think that's relevant for this. We don't guarantee the memory layout across breaking changes of the language as it is.